viam_cartographer.go

```diff
@@ -476,12 +515,13 @@ func terminateCartoFacade(ctx context.Context, cartoSvc *CartographerService) er
 type CartographerService struct {
 	resource.Named
 	resource.AlwaysRebuild
-	mu       sync.Mutex
-	SlamMode cartofacade.SlamMode
-	closed   bool
-	lidar    s.TimedLidarSensor
-	imu      s.TimedIMUSensor
-	subAlgo  SubAlgo
+	mu                 sync.Mutex
+	SlamMode           cartofacade.SlamMode
+	closed             bool
+	lidar              s.TimedLidarSensor
+	imu                s.TimedIMUSensor
+	movementSensorName string
```

Comment:
I needed to store the `movementSensorName`, since previously we determined it based on the imu name. Now that a movement sensor may be used as either an IMU or an odometer (or none of these), the name can not be reliably determined from the two sensors anymore.

Reply:
Can we extract the source imu/odom names from a merged movement sensor?

Reply:
I don't know if we can extract the source imu/odom names from the merged movement sensor. Either way, we can discuss this in the next PR where we're adding the odometer as a sensor.